lib/jsonapi/utils/list.ex

```diff
@@ -29,7 +31,9 @@ defmodule JSONAPI.Utils.List do
   end
 
   defp do_to_list_of_query_string_components({key, value}) when is_map(value) do
-    Enum.flat_map(value, fn {k, v} -> to_list_of_two_elem_tuple("#{key}[#{k}]", v) end)
+    Enum.flat_map(value, fn {k, v} ->
+      do_to_list_of_query_string_components({"#{key}[#{k}]", v})
```

Comment:
Annotation: This fix was just to make a recursive call, so that nested maps can be handled.